test/e2e/notifier/update/update.go

```diff
@@ -0,0 +1,207 @@
+				  "name": "test-pod-update-diff-false",
+				  "namespace": "test"
+				},
+				"spec": {
+				  "containers": [
+					{
+					  "name": "test-pod-container",
+					  "image": "tomcat:8.0"
+					}
+				  ]
+				}
+			  }
+			  `),
+			UpdateSetting: config.UpdateSetting{Fields: []string{"spec.containers[*].image"}, IncludeDiff: false},
+			Diff:          "spec.containers[*].image:\n\t-: tomcat:9.0.34\n\t+: tomcat:8.0\n",
+			ExpectedWebhookPayload: testutils.WebhookPayload{
+				EventMeta:   notify.EventMeta{Kind: "Pod", Name: "test-pod-update-diff-false", Namespace: "test", Cluster: "test-cluster-1"},
+				EventStatus: notify.EventStatus{Type: "update", Level: "warn", Reason: "", Error: "", Messages: []string(nil)},
+				Summary:     "Pod *test/test-pod-update-diff-false* has been updated in *test-cluster-1* cluster\n",
+			},
+		},
+		"create and update pod in configured namespace": {
+			// Diff message generated in Attachment
+			GVR:       schema.GroupVersionResource{Group: "", Version: "v1", Resource: "pods"},
+			Kind:      "Pod",
+			Namespace: "test",
+			Specs:     &v1.Pod{ObjectMeta: metav1.ObjectMeta{Name: "test-pod-update"}, Spec: v1.PodSpec{Containers: []v1.Container{{Name: "test-pod-container", Image: "tomcat:9.0.34"}}}},
+			ExpectedSlackMessage: testutils.SlackMessage{
+				Attachments: []slack.Attachment{{Color: "warning", Title: "v1/pods updated", Fields: []slack.AttachmentField{{Value: "Pod *test/test-pod-update* has been updated in *test-cluster-1* cluster\n```\nspec.containers[*].image:\n\t-: tomcat:9.0.34\n\t : tomcat:8.0\n\n```", Short: false}}, Footer: "BotKube"}},
+			},
+			Patch: []byte(`{
+				"apiVersion": "v1",
+				"kind": "Pod",
+				"metadata": {
+				  "name": "test-pod-update",
+				  "namespace": "test"
+				},
+				"spec": {
+				  "containers": [
+					{
+					  "name": "test-pod-container",
+					  "image": "tomcat:8.0"
+					}
+				  ]
+				}
+			  }
+			  `),
+			UpdateSetting: config.UpdateSetting{Fields: []string{"spec.containers[*].image"}, IncludeDiff: true},
+			Diff:          "spec.containers[*].image:\n\t-: tomcat:9.0.34\n\t+: tomcat:8.0\n",
+			ExpectedWebhookPayload: testutils.WebhookPayload{
+				EventMeta:   notify.EventMeta{Kind: "Pod", Name: "test-pod-update", Namespace: "test", Cluster: "test-cluster-1"},
+				EventStatus: notify.EventStatus{Type: "update", Level: "warn", Reason: "", Error: "", Messages: []string{"spec.containers[*].image:\n\t-: tomcat:9.0.34\n\t+: tomcat:8.0\n"}},
+				Summary:     "Pod *test/test-pod-update* has been updated in *test-cluster-1* cluster\n```\nspec.containers[*].image:\n\t-: tomcat:9.0.34\n\t+: tomcat:8.0\n\n```",
+			},
+		},
+	}
+
+	for name, test := range tests {
+		t.Run(name, func(t *testing.T) {
+
+			//checking if update operation is true
+			resource := fmt.Sprintf("%s/%s/%s", test.GVR.Group, test.GVR.Version, test.GVR.Resource)
+			if test.GVR.Group == "" {
+				resource = fmt.Sprintf("%s/%s", test.GVR.Version, test.GVR.Resource)
+			}
```

Comment:
We already have a helper function for this https://github.com/infracloudio/botkube/blob/develop/pkg/controller/controller.go#L270
Let's move GVR to gvrToString into utils, export and use that instead